source/proxy_strategy.c

```diff
@@ -1105,6 +1548,336 @@ struct aws_http_proxy_strategy_factory *aws_http_proxy_strategy_factory_new_tunn
 
 /******************************************************************************************************************/
 
+/*adaptive ntlm*/
+
+struct aws_http_proxy_strategy_factory_tunneling_ntlm {
+    struct aws_allocator *allocator;
+
+    struct aws_http_proxy_strategy_factory factory_base;
+
+    /* SA-TBI: add any factory state needed here */
+};
+
+struct aws_http_proxy_strategy_tunneling_ntlm {
+    struct aws_allocator *allocator;
+
+    struct aws_http_proxy_strategy strategy_base;
+
+    enum proxy_strategy_connect_state state;
+
+    /*
+     * SA-TBI: add any factory state needed here
+     *
+     * Likely things include response code (from the vanilla CONNECT) and the appropriate headers in
+     * the response
+     */
+};
+
+AWS_STATIC_STRING_FROM_LITERAL(s_proxy_authorization_header_ntlm_prefix, "NTLM ");
+
+/*
+ * Adds a proxy authentication header based on ntlm credential or response provided by user 
+ */
+static int s_add_ntlm_proxy_usertoken_authentication_header(
```

Comment:
Same issues as per the kerberos add_header functions

Reply:
Updated